### DISCUSSION TITLE
Noob here, wondering if I could get help with TrueNAS jail errors after following wiki tutorial

### DISCUSSION THREAD

**Discussion by Boofie0404** [FlareSolverr/FlareSolverr#1043] [Q&A]
Have followed the wiki tutorial to install FlareSolverr into a jail. However after running I get the following:

To change this login announcement, see motd(5).
root@FlareSolverr:~ # sysrc flaresolverr_enable="YES"
flaresolverr_enable:  -> YES
root@FlareSolverr:~ # sysrc flaresolverr_PORT="80"
flaresolverr_PORT:  -> 80
root@FlareSolverr:~ # service flaresolverr start
Starting flaresolverr.
root@FlareSolverr:~ #
root@FlareSolverr:~ # service flaresolverr status
flaresolverr is running as pid 30985.
root@FlareSolverr:~ #
root@FlareSolverr:~ # tail -f /var/log/flaresolverr/flaresolverr.log
2024-01-21 12:28:31 INFO     FlareSolverr 3.3.13
2024-01-21 12:28:31 INFO     Testing web browser installation...
2024-01-21 12:28:31 INFO     Platform: FreeBSD-13.1-RELEASE-p7-amd64-64bit-ELF
2024-01-21 12:28:31 INFO     Chrome / Chromium path: /usr/local/share/flaresolverr/_internal/chrome/chrome
2024-01-21 12:28:31 INFO     Chrome / Chromium major version: 120
2024-01-21 12:28:31 INFO     Launching web browser...
version_main cannot be converted to an integer
2024-01-21 12:28:33 INFO     FlareSolverr User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36
2024-01-21 12:28:33 INFO     Test successful!
Traceback (most recent call last):
  File "flaresolverr.py", line 122, in <module>
  File "bottle.py", line 3175, in run
  File "flaresolverr.py", line 121, in run
  File "waitress/__init__.py", line 13, in serve
  File "waitress/server.py", line 78, in create_server
  File "waitress/server.py", line 244, in __init__
  File "waitress/server.py", line 361, in bind_server_socket
  File "waitress/wasyncore.py", line 396, in bind
PermissionError: [Errno 13] Permission denied
[30985] Failed to execute script 'flaresolverr' due to unhandled exception!

Any help would be very much appreciated. Am very new to TrueNAS and trying to understand this so simple steps to solve would be very much appreciated!

**Comment by ilike2burnthing** [comment]
@FroggMaster

**Reply by TheSilasm8** [reply]
I'm getting the exact same error message. I've tried chmod on every created folder I could find.

**Reply by zehro** [reply]
Ditto here. Just tried to set it up today.

**Reply by ilike2burnthing** [reply]
Any luck with the instructions from before the last change?
https://github.com/FlareSolverr/FlareSolverr/wiki/Installing-FlareSolverr-in-FreeBSD,-FreeNAS-or-TrueNAS-jail/20b30f05b0434a8d59ff032f62b416ab98cef253

**Reply by TheSilasm8** [reply]
Figured it out I think. See my answer

**Comment by TheSilasm8** [comment]
I ran through the old version steps and I was getting this new error:

> root@FlareSolverr2:/usr/local/share/flaresolverr # service flaresolverr start
Starting flaresolverr.
root@FlareSolverr2:/usr/local/share/flaresolverr # service flaresolverr status
flaresolverr is not running.
root@FlareSolverr2:/usr/local/share/flaresolverr # tail -f /var/log/flaresolverr/flaresolverr.log
2024-01-27 16:09:13 INFO     Platform: FreeBSD-13.1-RELEASE-p9-amd64-64bit-ELF
2024-01-27 16:09:13 INFO     Chrome / Chromium path: /usr/local/share/flaresolverr/_internal/chrome/chrome
/usr/local/share/flaresolverr/_internal/chrome/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
2024-01-27 16:09:13 ERROR    Chrome / Chromium version not detected!
2024-01-27 16:28:52 INFO     FlareSolverr 3.3.13
2024-01-27 16:28:52 INFO     Testing web browser installation...
2024-01-27 16:28:52 INFO     Platform: FreeBSD-13.1-RELEASE-p9-amd64-64bit-ELF
2024-01-27 16:28:52 INFO     Chrome / Chromium path: /usr/local/share/flaresolverr/_internal/chrome/chrome
/usr/local/share/flaresolverr/_internal/chrome/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
2024-01-27 16:28:52 ERROR    Chrome / Chromium version not detected!

Figured out a few things:

1. Step 7.3: 
`cp /usr/local/bin/chrome ./dist/package/flaresolverr/chrome/chrome`
Should be
`cp /usr/local/bin/chrome ./dist/package/flaresolverr/_internal/chrome/chrome`

2. Step 7.3:
`#showing the ELF mark on the file`
`brandelf ./dist/package/flaresolverr/*.so.*`
`#change the ELF mark on the file (if is not 'Linux')`
`brandelf -t Linux ./dist/package/flaresolverr/*.so.*`
`#showing the result of the change`
`brandelf ./dist/package/flaresolverr/*.so.*`

None of these calls worked for me, I kept getting `brandelf: No match.`. I assume `*.so.*` is supposed to be regex to find flaresolverr, so I did this instead:
`#showing the ELF mark on the file`
`brandelf ./dist/package/flaresolverr/flaresolverr`
`#change the ELF mark on the file (if is not 'Linux')`
`brandelf -t Linux ./dist/package/flaresolverr/flaresolverr`
`#showing the result of the change`
`brandelf ./dist/package/flaresolverr/flaresolverr`
And that switched flaresolverr from 'FreeBSD' to 'Linux'.

I also made sure to skip calling `sysrc flaresolverr_PORT="80"`. Calling this also causes the ErrorNo 13: Permission Denied. If you called this, reset the port back to what it originally was: `sysrc flaresolverr_PORT="8191"`

That seemed to work for me.

**Reply by zehro** [reply]
Can confirm that it has launched! What a contribution~
![image](https://github.com/FlareSolverr/FlareSolverr/assets/7268988/4eb354cd-5cd7-4127-a6db-cf6fa31dc864)

Hopefully the FreeBSD guide can be updated soon for other confused users

**Reply by ilike2burnthing** [reply]
Please review updates to current version:
https://github.com/FlareSolverr/FlareSolverr/wiki/Installing-FlareSolverr-in-FreeBSD,-FreeNAS-or-TrueNAS-jail/_compare/9f5d27df8af2eabd9e085d8fac039a60e718ddcf...f7bfd1676c49b8627e73cb0d7b33ece9047c23a3

**Reply by TheSilasm8** [reply]
Seems good

**Reply by Boofie0404** [reply]
thankyou, you are all legends! Got it working now as well. My first time asking a question in the forums and was nice to see so many helpful people! Thanks!

**Comment by FroggMaster** [comment]
The first error you mentioned had already been corrected in my version of the WIKI regarding the copy path for Chrome. I'm not quite sure why it was still out of date when you were reviewing things unless some of my changes were still missing.  

My version of the WIKI has dropped the steps for brandelf as in my testing these changes were not necessary to get flaresolverr functional. 

After reading this comment thread, I've adjusted a few more things in my version of the WIKI relative to the final steps. It seems the wording was a little bit confusing but I believe that was supposed to be an example on how to change the port. Likely, your user needed additional permissions to run on port 80 or you have another service already running on port 80. I've removed that example and clarified the bottom section slightly. 

@ilike2burnthing 

Feel free to review 
https://github.com/FroggMaster/FlareSolverr/wiki/Installing-FlareSolverr-in-FreeBSD,-FreeNAS-or-TrueNAS-jail

**Reply by ilike2burnthing** [reply]
TheSilasm8 was referring to the old version linked here - https://github.com/FlareSolverr/FlareSolverr/discussions/1043#discussioncomment-8241068

I'll make those changes now, thanks.

**Reply by ilike2burnthing** [reply]
Oh, question. Since it was first created, throughout that wiki page the word `daemon` is being spelt `deamon`. Is this just a typo or is there something I'm missing?

**Reply by TheSilasm8** [reply]
When I initially went through the new wiki, I tried several things to try to figure out what was wrong before commenting but none of them worked. I ended up creating a fresh jail to run through the older wiki steps.

**Reply by FroggMaster** [reply]
@ilike2burnthing I suspect that is a typo from the older wiki, which I hadn't corrected.

**Reply by ilike2burnthing** [reply]
Ta, fixed both.
